### PR TITLE
Adding Copy and Move constructors in Dropout Layer

### PR DESCRIPTION
**Pull request by NippunSharma**
This pull request is in reference to #2625 
It adds the copy and move constructors and assignment operators to the Dropout layer.

### CODE REVIEW

**Comment by NippunSharma** [comment]
Please ignore the above mentioned PR. I did not know that github automatically links the PR's we mention.

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 1]
```diff
@@ -153,6 +153,62 @@ TEST_CASE("CheckCopyMovingVanillaNetworkTest", "[FeedForwardNetworkTest]")
...
+   *  Bias       |  Bias       |
+   *  Layer      |  Layer      |
+   * +-----+     | +-----+     |
+   * |     |     | |     |     |
+   * |     +-----+ |     +-----+
+   * |     |       |     |
+   * +-----+       +-----+
+   */
+
+  FFN<NegativeLogLikelihood<> > *model = new FFN<NegativeLogLikelihood<> >;
```
Comment (zoq) [inline src/mlpack/tests/feedforward_network_test.cpp:190]: We should delete the model at the end.

Comment (NippunSharma) [reply src/mlpack/tests/feedforward_network_test.cpp:190]: I think that the model is deleted already inside the CheckCopyFunction and CheckMoveFunction

**Comment by NippunSharma** [comment]
@zoq is my observation correct or should I delete the model at the end ?

**Comment by NippunSharma** [comment]
Hi @zoq , what do you think about this ?

File: src/mlpack/tests/feedforward_network_test.cpp (hunk 1 above)
Comment (zoq) [reply src/mlpack/tests/feedforward_network_test.cpp:190]: Sorry for the slow response, you are correct - https://github.com/mlpack/mlpack/blob/f3e3735371d54c37fe191dd17e2063652cab5909/src/mlpack/methods/ann/ffn_impl.hpp#L49-L50

**Review by zoq — APPROVED** [review]
Looks good to me, thanks for putting this together.

**Comment by NippunSharma** [comment]
@zoq can you merge this if there are no more changes ?

**Comment by zoq** [comment]
Great, thanks again for contribution.

**Pull request merged**